Samples/Widgets/SingleWidget/cpp-win32-packaged/SampleWidgetProviderApp/WidgetProvider.cpp

```diff
@@ -0,0 +1,205 @@
+
+	winrt::WidgetUpdateRequestOptions updateOptions{ widgetId };
+	updateOptions.Template(GetTemplateForWidget());
+	updateOptions.Data(GetDataForWidget(widgetId));
+	// You can store some custom state in the widget service that you will be able to query at any time.
+	updateOptions.CustomState(GetStateForWidget(widgetId));
+	// Update the widget
+	winrt::WidgetManager::GetDefault().UpdateWidget(updateOptions);
+}
+
+// Handle the DeleteWidget call. This is notifying you that
+// you don't need to provide new content for the given WidgetId
+// since the user has unpinned the widget or it was deleted by the Host
+// for any other reason.
+void WidgetProvider::DeleteWidget(winrt::hstring const& widgetId, winrt::hstring const& /*customState*/)
+{
+	m_runningWidgets.erase(widgetId);
+}
+
+// Handle the OnActionInvoked call. This function call is fired when the user's
+// interaction with the widget resulted in an execution of one of the defined
+// actions that you've indicated in the template of the Widget.
+// For example: clicking a button or submitting input.
+void WidgetProvider::OnActionInvoked(winrt::WidgetActionInvokedArgs actionInvokedArgs)
+{
+	auto verb = actionInvokedArgs.Verb();
+	auto widgetContext = actionInvokedArgs.WidgetContext();
+	auto widgetId = widgetContext.Id();
+	if (auto&& widget = FindRunningWidget(widgetId))
+	{
+		if (verb == L"inc")
+		{
+			widget->m_clickCount++;
+			// Generate template/data you want to send back
+			// The template has not changed so it does not neeed to be updated
+			auto widgetData = GetDataForWidget(widgetId);
+
+			// Build update options and update the widget
+			winrt::WidgetUpdateRequestOptions updateOptions{ widgetId };
+			updateOptions.Data(widgetData);
+			updateOptions.CustomState(GetStateForWidget(widgetId));
+
+			winrt::WidgetManager::GetDefault().UpdateWidget(updateOptions);
+			return;
+		}
+		throw winrt::hresult_invalid_argument{ L"OnActionInvoked invoked for widget with id: " + widgetId + L" and unknown verb: " + verb};
+	}
+	throw winrt::hresult_invalid_argument{ L"OnActionInvoked invoked for unknown widget with id: " + widgetId };
+}
+
+// Handle the WidgetContextChanged call. This function is called when the context a widget
+// has changed. Currently it only signals that the user has changed the size of the widget.
+// There are 2 ways to respond to this event:
+// 1) Call UpdateWidget() with the new data/template to fit the new requested size.
+// 2) If previously sent data/template accounts for various sizes based on $host.widgetSize - you can use this event solely for telemtry.
+void WidgetProvider::OnWidgetContextChanged(winrt::WidgetContextChangedArgs contextChangedArgs)
+{
+	auto widgetContext = contextChangedArgs.WidgetContext();
+	auto widgetId = widgetContext.Id();
+	if (auto&& widget = FindRunningWidget(widgetId))
+	{
+		// Optionally: if the data/template for the new context is different
+		// from the previously sent data/template - send an update.
+	}
```

Comment:
This needs a `return;` here in the loop - people copy & paste our samples into their apps, and as written this function always throws, even if the app didn't care about the change in context.

Reply:
Added a return statement